processing/src/main/java/io/druid/query/extraction/RegexDimExtractionFn.java

```diff
@@ -86,11 +86,8 @@ public RegexDimExtractionFn(
   @Override
   public String apply(String dimValue)
   {
-    if (dimValue == null) {
-      return null;
-    }
-    String retVal;
-    Matcher matcher = pattern.matcher(dimValue);
+    final String retVal;
+    final Matcher matcher = pattern.matcher(Strings.nullToEmpty(dimValue));
```

Comment:
same here, if dimValue is `null`, if there is a match, then matcher.group(1) will always be empty, so we can skip the matching, and if there is no match, then we return null/replaceMissingValueWith anyways. So we can skip the nullToEmpty and matching entirely.